hips/hip-9999.md

```diff
@@ -0,0 +1,79 @@
+
+The second part of this proposal would include the tarball digest of a chart in the metadata. This aspect is less fleshed out in code yet, but would be a nice alternative for installing a local chart from a tarball when the repository URL would not be known. The reason this is useful is because it can be used when interacting with other third-party databases, such as artifact-hub.
+
+## Motivation
+
+The primary motivation is to have a helm-native way to track where a chart came from. Many of the motivations for this feature can be found in this issue: https://github.com/helm/helm/issues/4256. To note a couple:
+- An admin would like to know where a chart came from only by querying the cluster itself. Currently the only way to know where a chart came from is to keep track of it externally.
+- Third-party tooling to that provides recommendations on helm upgrades can query the cluster to get the repository URL to aid in this recommendation. See [nova](https://github.com/FairwindsOps/nova) as an example.
+- This metadata field could also populate output for the `helm ls` command in the future. That may be outside the scope of this particular feature.
+
+## Rationale
+
+In order to query for this in the cluster, the repository URL will need to be somewhere in the release object. The chart metadata portion of the release object already contains similar data and makes the most sense for this purpose.
+
+## Specification
+
+Nothing would change from the user's perspective. With some modifications in the current code it's possible to add this metadata into the release object transparently.
+
+The current idea for this implementation is to add a field in both the `ChartDownloader` and `Metadata` (in the `pkg/chart/metadata.go` file) structs to hold the repository URL. The Downloader is the logical place to derive the proper URL and can then that storage location can populate the Metadata field.
+
+More specifically, the URL is included in the local repository index file for a locally named repository. For instance, running `helm repo add [NAME] [URL]` would add the repository to the index file with the `[URL]` passed as the real location to download charts. In the helm code, the downloader reads this file to find the URL which is what gets bubbled up to the release object.
+
+If for any reason this data can not be gathered at install/upgrade time, then that particular field would be omitted from the release object metadata.
+
+The chart metadata object would have new fields that looked like so:
+
+```
+"repoURL": "https://charts.fairwinds.com/stable"
+"tarballDigest": "sha256:f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9"
+```
+
+And an OCI compliant repository would look like this (example is using ECR):
+
+```
+"repoURL": "oci://${ACCOUNT_NUMBER}.dkr.ecr.us-east-1.amazonaws.com/goldilocks"
+"tarballDigest": "sha256:f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9f9a9"
+```
```

Comment:
How would this information be accessed?  Is it meant to be easily accessible by users?  Is it sufficient to add it to the metadata and leave it up to users to figure out how to access it using `kubectl` or some other means?

Reply:
I leave it up to the users here. If it's decided that it should be made a part of `helm ls` or something later, I think that can be a separate request/HIP.